src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1108,6 +1112,9 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 			combinedShellArgs.push(commandLine);
 			shellLaunchConfig.args = windowsShellArgs ? combinedShellArgs.join(' ') : combinedShellArgs;
 			if (task.command.presentation && task.command.presentation.echo) {
+				// this may have been set to the first folder in a multi-root repo
```

Comment:
ideally this wouldn't be needed, so have to find an earlier place to resolve the variables and update the workspace folder

Reply:
Thoughts @alexr00 ?